### PR TITLE
Promote driver version to v0.2.0.beta

### PR DESCRIPTION
**Pull request by davidz627**
/hold
until release cut, image pushed, blog post ready.

/assign @msau42 
for preliminary review

### CODE REVIEW

File: deploy/kubernetes/stable/node.yaml [hunk 1]
```diff
@@ -16,52 +16,56 @@ spec:
...
           volumeMounts:
             - name: kubelet-dir
               mountPath: /var/lib/kubelet
               mountPropagation: "Bidirectional"
             - name: plugin-dir
               mountPath: /csi
             - name: device-dir
-              mountPath: /host/dev
+              mountPath: /dev
+            # The following mounts are required to trigger host udevadm from container
```
Comment (msau42) [inline deploy/kubernetes/stable/node.yaml]: didn't this one already merge?

Comment (davidz627) [reply deploy/kubernetes/stable/node.yaml]: moved to stable

**Comment by davidz627** [comment]
Note to self: *Do not unhold without pulling in snapshotter yaml changes as well"

**Comment by davidz627** [comment]
Blocked on a couple things:
* Snapshot PR: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pull/120
* Secure build process (tracked internally)
* Migration support PR: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pull/129
* Blog post (tracked internally)

**Comment by davidz627** [comment]
/hold cancel
@msau42 ready for review

File: Makefile [hunk 2]
```diff
@@ -18,8 +18,8 @@
 STAGINGIMAGE=${GCE_PD_CSI_STAGING_IMAGE}
 STAGINGVERSION=${GCE_PD_CSI_STAGING_VERSION}
 
-PRODIMAGE=gcr.io/google-containers/volume-csi/gcp-compute-persistent-disk-csi-driver
-PRODVERSION=v0.1.0.alpha
+PRODIMAGE=gcr.io/k8s-image-staging/volume-csi/gcp-compute-persistent-disk-csi-driver
```
Comment (msau42) [inline Makefile]: We should probably use the official google repo for this and not a staging repo

File: deploy/kubernetes/stable/controller.yaml [hunk 3]
```diff
@@ -15,12 +15,25 @@ spec:
     spec:
       serviceAccount: csi-controller-sa
       containers:
+        - name: csi-snapshotter
+          imagePullPolicy: Always
+          image: quay.io/k8scsi/csi-snapshotter:v0.4.0
```
Comment (msau42) [inline deploy/kubernetes/stable/controller.yaml]: should we just use 4.1 for everything?

File: deploy/kubernetes/stable/controller.yaml [hunk 4]
```diff
@@ -15,12 +15,25 @@ spec:
...
+              mountPath: /csi
+
         - name: csi-provisioner
           imagePullPolicy: Always
-          image: quay.io/k8scsi/csi-provisioner:v0.2.0
+          image: quay.io/k8scsi/csi-provisioner:v0.4.1
           args:
             - "--v=5"
-            - "--provisioner=csi-gce-pd"
+            - "--provisioner=com.google.csi.gcepd"
```
Comment (msau42) [inline deploy/kubernetes/stable/controller.yaml]: this can be removed now right?

File: deploy/kubernetes/stable/controller.yaml (hunk 3 above)
Comment (msau42) [reply deploy/kubernetes/stable/controller.yaml]: Snapshot is alpha, should we include it in stable?

Comment (davidz627) [reply deploy/kubernetes/stable/controller.yaml]: good point, i've removed it from stable

File: deploy/kubernetes/stable/controller.yaml (hunk 4 above)
Comment (davidz627) [reply deploy/kubernetes/stable/controller.yaml]: oh yes, thats true.

**Comment by davidz627** [comment]
@msau42 comments addressed

File: deploy/kubernetes/stable/controller.yaml [hunk 5]
```diff
@@ -17,10 +17,10 @@ spec:
       containers:
         - name: csi-provisioner
           imagePullPolicy: Always
-          image: quay.io/k8scsi/csi-provisioner:v0.2.0
+          image: quay.io/k8scsi/csi-provisioner:v0.4.1
```
Comment (msau42) [inline deploy/kubernetes/stable/controller.yaml]: can you open up a new issue to track using gke officially released public images for sidecars and driver? (the staging ones are not publicly accessible)

Comment (davidz627) [reply deploy/kubernetes/stable/controller.yaml]: https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/issues/135

**Pull request merged**